eventlet/tpool.py

```diff
@@ -333,4 +338,12 @@ def killall():
 
 def set_num_threads(nthreads):
     global _nthreads
-    _nthreads = nthreads
+    old_nthreads, _nthreads = _nthreads, nthreads
+
+    if _setup_already:
+        if old_nthreads > nthreads:
+            import warnings
+            warnings.warn("Dynamically reducing the number of threads is not "
```

Comment:
Please remove this warning.

Reply:
That doesn't sound quite right, if caller makes a request thinking it will dynamically changed and we are not going to fulfill it, shouldn't we issue a warning?  (I see that I forgot to pass the "RuntimeWarning" parameter).

Reply:
Yes, breaking expected behavior is not right, I agree. Here's why I think there should be no warning. If you have little time, just skip to solutions. :-)

First, and most important, warning still doesn't deliver expected behavior. Second, in this particular case, I believe, warning is unlikely to find a human that cares or will change something. Alternatives down below.

There will be two kinds of human readers of warning:
- user of eventlet, writer of openstack or other software. It is target audience for warning indeed. Though blunt and plain `ValueError('must increase only')` would be far more effective here.
- final user of software. It may not be noticed or mildly annoy user who can't fix "unsupported reducing of what is it even talking about" (constructive criticism below). He might look for a way to disable warning and find a stupid ugly recipe to disable all warnings. This is bad and unfortunately, happens more often than I'd like to think. Python is next PHP I hope it conveys my point without hurting feelings. Coders (as opposed to software engineers) are the workhorse of IT industry and caring for them means making libraries that just work in the best possible way, without much distraction.

### Solution options

Basically, either fulfill expectation or maintain warning silence for really important things. :-)
- (best) actually implement pool scaling down to requested size. I can take this task.
- (acceptable) just remove warning and assume grow-only is acceptable (for a while) until scaling down is not implemented. Really, the way **global** tpool is used, grow-only can hardly make harm.
- (worst) rename `set_num_threads` to `increase_pool_size` with deprecation warning on former ("301 redirect" if HTTP is your second language). This removes wrong expectation.

FYI, don't go with this in this case, but good stuff in general. Separate warning class to ease disabling. You have unlimited room for description text. Don't assume reader knows what is thread pool. Explain how serious is the problem and what could be done. Link to case history. Example: https://github.com/eventlet/eventlet/blob/f1b63abd6db186c978077499f9670600da599d1a/eventlet/convenience.py#L53-L76